git-make-release

```diff
@@ -98,28 +75,38 @@ elif [ "${LAST_VERSION_ARRAY[1]}" != "${MINOR}" ]; then
     echo "*** Warning: Minor release, but patch is not zero!"
   fi
 fi
+if [[ ${CURRENT_BRANCH} != master ]]; then
+  RELEASE_TYPE="BACKPORT"
+fi
 
 echo "Release type: ${RELEASE_TYPE}"
 
-# ask whether to proceed
-echo ""
-echo -n "Do you want to proceed with creating the tag (y/N)? "
-read ANSWER
-if [ "$ANSWER" != "y" ]; then
-  echo "Aborted."
-  rm -f $CHANGES
-  exit 1
-fi
+# obtain the change log since the latest tag
+CHANGES=`mktemp`
+git log ${LAST_VERSION}..HEAD --pretty=%s | grep -v '^project-template:' | grep -v '(project-template): '           \
```

Comment:
I am wondering if '^Merge pull request ' and '^Merge branch ' might be too generic and filter too much. This script is also used on projects which allow merging into the master and might not follow the conventional commit, so it might filter important commit messages like "Merge branch my-important-feature".

How about ```grep -iv "project.template"```?

Reply:
The idea is also to filter merge commit messages which never contain any information I want to see in the release notes. The message "Merge branch my-important-feature" should not be listed in any release notes, instead the commits which are merged should be found and included.

Do you think this idea is bad?

Reply:
Chances are that the commit should actually read
"feat: my important feature" and the rest of the probably uncleaned history should have been squashed with our new workflow.

I would leave this out. If you follow conventional commit, there are no such commit messages. Hence, for your projects and the core libraries it makes no difference. And if there is such a message you might not want to filter it because it follows a different scheme and might contain important information.

Reply:
One reason for conventional commits and the merging rules (no merge commit) was to simplify the release note generation. If a commit is called "Merge branch my-important-feature" but should be actually named "feat: my important feature", changes are close to 100% I would throw this commit out manually when generating the release notes already now (without the proposed change). An ordinary (non-merge) commit titled "Merge branch..." would be very very misleading under any circumstance, so I really think this should never ever be done.

On the other hand, we get merge commits virtually exclusively when merging the project-template (because they are disallowed normally), and in that case we get at least on Github typically two of them directly following each other. The exact format of the merge commit message depends on the git client used to do the merge, I think this might actually vary from version to version and configuration to configuration slightly. Not catching all of them here will be very inconvenient and also will just result in everyone removing all lines in the generated release notes starting with "Merge branch" anyway.

Even when recognising the situation of a failed squash merge which ends up as a merge commit with many individual commits (with garbage titles), the presence of the merge commit is no longer helpful due to the absence of the graph information. There is basically no way of understanding which commit belongs to the merge commit. Also, the only way of recognising this situation would be to find all the garbage commit titles, in which case the maintainer can just look into the history manually with another tool and correct the release notes.

As a conclusion I would keep this filtering in here as is. I see no benefit but a lot of inconvenience of not removing the merge commit messages from the release notes automatically. The goal in the end is that the release notes can often be used as generated without modification, or maybe after minimal editing only (like sorting by importance).